### PR TITLE
mem/libminimal.la to facilitate helper reuse of convenience libs

### PR DESCRIPTION
**Pull request by rousskov**
Squid helpers/tools cannot reuse features like SBuf because, in part,
the corresponding convenience libraries depend on libmem.la which drags
in heavy dependencies, including Store and time-based Event modules. The
new mem/libminimal convenience library implements enough of src/mem/
APIs to make the new memory library usable by helpers/tools without
causing a dependency explosion.

Do not use tests/stub_libmem.cc in the deployed/non-test squidclient,
cachemgr.cgi, and pinger programs.

### CODE REVIEW

**Comment by rousskov** [comment]
This PR depends on and contains PR #1003 changes (the first two commits on this PR branch). Unfortunately, those changes cannot be removed without creating conflicts with master. When PR #1003 is merged, this PR will be adjusted to remove those merged changes. Meanwhile, the last PR branch commit -- [6944350](https://github.com/squid-cache/squid/pull/1004/commits/6944350f64c67579896ef2f2ef21b35a169be199) -- can be used for the initial review of changes specific to this PR.

**Review by yadij — COMMENTED** [review]
One bit of polish to update during the conflict resolution.

File: src/mem/Makefile.am [hunk 1]
```diff
@@ -8,11 +8,19 @@
...
-noinst_LTLIBRARIES = libmem.la
+noinst_LTLIBRARIES = \
+	libmem.la \
+	libminimal.la
 
+noinst_HEADERS = \
+	AllocatorProxy.h \
+	Sensitive.h \
+	forward.h
+
```
[the remark this answers is not in the document]
Comment (rousskov) [reply src/mem/Makefile.am:19]: Fixed at https://github.com/squid-cache/squid/commit/37ebdd5c915177f47a4de57c93e7952df7364b77

File: src/mem/minimal.cc [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+ */
+
+#include "squid.h"
+#include "mem/AllocatorProxy.h"
+#include "mem/forward.h"
+
+/// The number of currently alive objects (poor man's meter.alloc=meter.inuse).
+/// Technically, this is supposed to be a per-allocator statistics, but
+/// AllocatorProxy is not a MemAllocator so we maintain a global counter
+/// instead. We probably do not have to maintain this statistics at all.
```
[the remark this answers is not in the document]
Comment (rousskov) [reply src/mem/minimal.cc:16]: You are right that this information is essential for mgr:mem, but mgr:mem does not (and will never) use libminimal to which this specific code belongs. This code and libminimal is for programs other than sbin/squid, programs that do not support cache manager reports. This PR does not change mgr:mem coverage.

**Comment by rousskov** [comment]
> When PR 1003 is merged, this PR will be adjusted to remove those merged changes.

Done at [f07c046](https://github.com/squid-cache/squid/pull/1004/commits/f07c0467fc166fbaf61f2d31ed9e09ef0b40c61c).

**Review by yadij — APPROVED** [review]

**Pull request closed without merging**